analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/SecureRandomSeedsShouldNotBePredictable.cs

```diff
@@ -145,19 +165,40 @@ private static ProgramState ProcessSecureRandomSetSeed(ProgramState state, IInvo
             {
                 constraint = value;
             }
-            return state.SetSymbolConstraint(invocation.Instance.TrackedSymbol(state), constraint);
+            return state.SetSymbolConstraint(instanceSymbol, constraint);
         }
         return null;
+
+        bool IsSetSeed() =>
+            invocation.TargetMethod.Name == "SetSeed"
+            && IsSecureRandom(invocation);
+
+        bool IsAddSeedMaterial() =>
+            invocation.TargetMethod.Name == "AddSeedMaterial"
+            && invocation.Instance is { } instance
+            && instance.Type.Is(KnownType.Org_BouncyCastle_Crypto_Prng_IRandomGenerator);
     }
 
-    private ProgramState ProcessSecureRandomNext(ProgramState state, IInvocationOperationWrapper invocation)
+    private ProgramState ProcessNextMethods(ProgramState state, IInvocationOperationWrapper invocation)
```

Comment:
This used to track `SecureRandom.NextXXX`, now it ALSO tracks `IRandomGenerator.NextBytes`